WordPress/Classes/ViewRelated/What's New/Data store/AnnouncementsStore.swift

```diff
@@ -45,16 +48,26 @@ class RemoteAnnouncementsStore: AnnouncementsStore {
         }
     }
 
+    init(cache: AnnouncementsCache) {
+        self.cache = cache
+    }
+
     func getAnnouncements() {
-        let service = AnnouncementServiceRemote(wordPressComRestApi: api)
         state = .loading
+        if let announcements = cache.announcements {
+            self.state = .ready(announcements)
+            return
+        }
+
+        let service = AnnouncementServiceRemote(wordPressComRestApi: api)
         service.getAnnouncements(appId: Identifiers.appId,
                                  appVersion: Identifiers.appVersion,
                                  locale: Locale.current.identifier) { result in
 
             switch result {
             case .success(let announcements):
                 self.state = .ready(announcements)
+                self.cache.announcements = announcements
```

Comment:
Should these two statements be the other way round? Setting the state emits the change event, but so the cached announcements should probably be in place before that happens.

Reply:
So what sets the state is only the `getAnnouncements` method.

The flow should be:

- if there are cached values, use them and return
- if not:
  - fetch them from the backend and, as soon as you have them, use them (do not wait to save them on disk)
  - "at the same time", save them on disk, but don't let this operation block your process

So I believe that the change here would be to dispatch the save-on-disk asynchronously on the global queue